watchtower-plugin/src/retrier.rs

```diff
@@ -85,87 +95,161 @@ impl Retrier {
+                    .unwrap()
+                    .load_appointment(locator)
+                    .unwrap();
+
+                match add_appointment(
+                    tower_id,
+                    &net_addr,
+                    proxy.clone(),
+                    &appointment,
+                    &cryptography::sign(&appointment.to_vec(), &user_sk).unwrap(),
+                )
+                .await
+                {
+                    Ok((slots, receipt)) => {
+                        self.pending_appointments
+                            .lock()
+                            .unwrap()
+                            .get_mut(&tower_id)
+                            .unwrap()
+                            .remove(&locator);
+                        let mut wt_client = self.wt_client.lock().unwrap();
+                        wt_client.add_appointment_receipt(
+                            tower_id,
+                            appointment.locator,
+                            slots,
+                            &receipt,
+                        );
+                        wt_client.remove_pending_appointment(tower_id, appointment.locator);
+                        log::debug!("Response verified and data stored in the database");
+                    }
+                    Err(e) => {
+                        match e {
+                            AddAppointmentError::RequestError(e) => {
+                                if e.is_connection() {
+                                    log::warn!(
+                                        "{} cannot be reached. Tower will be retried later",
+                                        tower_id,
+                                    );
+                                    return Err(Error::transient("Tower cannot be reached"));
+                                }
                             }
-                            _ => {
-                                log::warn!(
-                                    "{} rejected the appointment. Error: {}, error_code: {}",
-                                    tower_id,
-                                    e.error,
-                                    e.error_code
-                                );
-                                // We need to move the appointment from pending to invalid
-                                // Add itn first to invalid and remove it from pending later so a cascade delete is not triggered
-                                let mut wt_client = self.wt_client.lock().unwrap();
-                                wt_client.add_invalid_appointment(tower_id, &appointment);
-                                wt_client.remove_pending_appointment(tower_id, appointment.locator);
+                            AddAppointmentError::ApiError(e) => match e.error_code {
+                                errors::INVALID_SIGNATURE_OR_SUBSCRIPTION_ERROR => {
+                                    log::warn!("There is a subscription issue with {}", tower_id);
```

Comment:
There is a case that would mis-classify a tower state, goes as follows:
1- tower A was temp unreachable and sent to be retried
2- tower A is waiting its turn to be retried and more appointments accumulate
3- tower A started being retried and at halfway through sending all the pending appointment, it started giving back subscription errors.
Such a tower would have temp unreachable status, and then unreachable but never classified as subscription error.